src/docfx/build/Build.cs

```diff
@@ -63,7 +63,13 @@ internal static class Build
 
                 bool ShouldBuildFile(Document file)
                 {
-                    return file.ContentType != ContentType.Unknown && fileListBuilder.TryAdd(file);
+                    if (file.ContentType == ContentType.Unknown)
+                        return false;
+                    if (!fileListBuilder.TryAdd(file))
+                        return false;
+                    if (file.ContentType == ContentType.Resource && !file.Docset.Config.Output.CopyResources)
+                        return false;
```

Comment:
Move this logic to BuildResource?

Reply:
yeah, the only downside is progress, it will immediately jump from 30% to 100%